app/controllers/v1/ControllerUtils.scala

```diff
@@ -103,38 +107,62 @@ trait ControllerUtils extends Controller with StrictLogging with Properties with
     case ex => InternalServerError(errAsJson(ex.toString, ex.getCause.toString))
   }
 
+  protected def louSearch(baseUrl: Uri, id: String, period: String)(implicit ws: RequestGenerator): Future[Result] = id match {
+    case validId if (id.length >= MINIMUM_KEY_LENGTH) => {
+      LOGGER.info(s"Sending request to ${baseUrl.toString} to retrieve Unit Links")
+      ws.singleGETRequest(baseUrl.toString) map {
+        case response if response.status == OK => {
```

Comment:
This looks like an Incomplete match.  This will fail at runtime with a MatchError if the server responds with a 500 for example.  This should instead be handled gracefully as part of the "flow" of the application.